src/featureforest/utils/data.py

```diff
@@ -1,106 +1,259 @@
+from typing import Tuple, List, Optional
+
 import numpy as np
+from numpy import ndarray
 import torch
 import torch.nn.functional as F
+from torch import Tensor
 
 
-# IMAGE_PATCH_SIZE = 512
-# TARGET_PATCH_SIZE = 128
+def get_patch_size(img_height: float, img_width: float) -> int:
+    """Calculate the patch size given image dimensions.
 
+    Args:
+        img_height (int): image height
+        img_width (int): image width
 
-def patchify(imgs, patch_size, target_size):
+    Returns:
+        int: patch size
     """
-    imgs: (B, C, H, W)
-    out: (B*N, C, patch_size, patch_size)
+    patch_size = 512
+    img_min_dim = min(img_height, img_width)
+    while img_min_dim - patch_size < 128:
```

Comment:
Changed in commit 2581935.  
Now, if the minimum image dim is less than the default patch size (256), it will choose the power of two number that is smaller than image dim.  
Otherwise, if the number of patches are less than **two**, the `patch_size` will be divided by 2 in a while loop to get the proper `patch_size`.